functions/src/definitions/batchJobs.js

```diff
@@ -68,14 +75,51 @@ async function checkConversationSessionExpiring(context) {
       .where("timestamp", ">", windowEnd) //and all those later than 24 hours ago
       .get()
     const promisesArr = unrepliedInstances.docs.map(async (doc) => {
-      await respondToInstance(doc, true)
+      return respondToInstance(doc, true)
     })
     await Promise.all(promisesArr)
   } catch (error) {
     functions.logger.error("Error in checkConversationSessionExpiring:", error)
   }
 }
 
+async function interimPromptHandler(context) {
+  try {
+    const db = admin.firestore()
+    const dayAgo = Timestamp.fromDate(
+      new Date(Date.now() - 24 * 60 * 60 * 1000)
+    )
+    const halfHourAgo =
+      runtimeEnvironment.value() === "PROD"
+        ? Timestamp.fromDate(new Date(Date.now() - 30 * 60 * 1000))
+        : Timestamp.fromDate(new Date())
+    const thresholds = await getThresholds()
+    const eligibleInstances = await db
+      .collectionGroup("instances")
+      .where("isReplied", "==", false) //match all those that haven't been replied to
+      .where("timestamp", ">", dayAgo) //and came in later than 24 hours ago
+      .where("timestamp", "<=", halfHourAgo) //but also at least 30 minutes ago
+      .where("isInterimPromptSent", "==", null) //and for which we haven't sent the interim yet
+      .get()
+    const promisesArr = eligibleInstances.docs.map(async (doc) => {
+      const parentMessageRef = doc.ref.parent.parent
+      const voteCount = await getCount(parentMessageRef, "responses")
+      if (
+        voteCount >= runtimeEnvironment.value() === "PROD"
+          ? thresholds.sendInterimMinVotes
+          : 1
+      ) {
+        return sendInterimPrompt(doc)
+      } else {
+        return Promise.resolve()
```

Comment:
I see `return Promise.resolve()` has been used in several places. actually if the block is already an async function, you can just `return`